src/comm.cc

```diff
@@ -1024,15 +1030,7 @@ comm_remove_close_handler(int fd, AsyncCall::Pointer &call)
 static void
 commSetNoLinger(int fd)
 {
-
-    struct linger L;
-    L.l_onoff = 0;      /* off */
-    L.l_linger = 0;
-
-    if (setsockopt(fd, SOL_SOCKET, SO_LINGER, (char *) &L, sizeof(L)) < 0) {
-        int xerrno = errno;
-        debugs(50, DBG_CRITICAL, MYNAME << "FD " << fd << ": " << xstrerr(xerrno));
-    }
+    commConfigureLinger(fd, OnOff::off);
     fd_table[fd].flags.nolinger = true;
```

Comment:
Forgot to write this in the earlier review: It is difficult for me to justify keeping commSetNoLinger() method, especially since flags.nolinger is never consulted AFAICT. Please replace this method calls with commConfigureLinger() calls if you agree.

Reply:
If we remove flags.nolinger=true from here (replacing commSetNoLinger() with commConfigureLinger()), there may be questions why we still keep such assignment in comm_import_opened() - i.e., shouldn't we remove this flag at all as unused. Also there is one (but commented out) commSetNoLinger() in ipc_win32.cc - we would need to modify that comment, creating extra noise. So I would keep this function for now.

Reply:
> shouldn't we remove this flag at all as unused.

We should. I should have said that explicitly.

> Also there is one (but commented out) commSetNoLinger() in ipc_win32.cc - we would need to modify that comment, creating extra noise. So I would keep this function for now.

I still cannot justify adding a second function that does the same thing as the existing (poorly named) function, despite the noise required to remove the existing function.

I see two options for going forward:

A: Remove flags.nolinger as unused in a prerequisite PR. After that, it becomes just an argument about naming the function correctly. Hopefully, we will be allowed to remove "no" from a function name despite the noise.

B: Propose to do everything in one official PR and split that PR if others insist on that.

I recommend B, but it is a weak recommendation. Your call.

Reply:
> I recommend B

Done at 883a74a. I agree that we should try posting all in one PR (since the changes look pretty trivial).